xmppserver/src/main/java/org/jivesoftware/openfire/session/LocalSession.java

```diff
@@ -416,12 +427,18 @@ public void process(Packet packet) {
 
     @Override
     public void deliverRawText(String text) {
-        if ( conn == null )
-        {
-            Log.debug( "Unable to deliver raw text in session, as its connection is null. Dropping: " + text );
-            return;
+        lock.lock();
```

Comment:
Don't think this lock/unlock is necessary

Reply:
done